src/libraries/System.ComponentModel.TypeConverter/tests/ReflectTypeDescriptionProviderTests.cs

```diff
@@ -0,0 +1,164 @@
+        {
+            object provider = CreateReflectTypeDescriptionProviderInstance();
+            AttributeCollection attributeCollection = provider.GetType().GetMethod("GetAttributes", BindingFlags.Instance | BindingFlags.NonPublic)
+                .Invoke(provider, new[] { typeof(TestClass1) }) as AttributeCollection;
+            Assert.NotEmpty(attributeCollection);
+            IEnumerable<Attribute> attributes = attributeCollection.Cast<Attribute>();
+            Attribute attribute =  Assert.Single(attributes);
+            Assert.IsType<DescriptionAttribute>(attribute);
+            Assert.DoesNotContain(attributeCollection.Cast<Attribute>(), attr => attr.GetType() == typeof(ComVisibleAttribute));
+            Assert.DoesNotContain(attributeCollection.Cast<Attribute>(), attr => attr.GetType() == typeof(GuidAttribute));
+            Assert.DoesNotContain(attributeCollection.Cast<Attribute>(), attr => attr.GetType() == typeof(InterfaceTypeAttribute));
+        }
+
+        [Fact]
+        public void GetExtenderProviders_ReturnResultFromContainerComponents_WhenComponentSiteServiceNull()
+        {
+            object provider = CreateReflectTypeDescriptionProviderInstance();
+            using TestComponent testComponent = new TestComponent();
+            testComponent.Site = new TestSiteWithoutService();
+            testComponent.Disposed += (object obj, EventArgs args) => { };
+            IExtenderProvider[] extenderProviders = provider.GetType().GetMethod("GetExtenderProviders", BindingFlags.Instance | BindingFlags.NonPublic)
+                .Invoke(provider, new[] { testComponent }) as IExtenderProvider[];
+            IExtenderProvider result = Assert.Single(extenderProviders);
+            Assert.IsType<ComponentExtendedProvider>(result);
+        }
+
+        [Fact]
+        public void GetExtenderProviders_ReturnResultFromComponentSiteService_WhenComponentSiteServiceNotNull()
+        {
+            object provider = CreateReflectTypeDescriptionProviderInstance();
+            using TestComponent testComponent = new TestComponent();
+            testComponent.Site = new TestSiteWithService();
+            testComponent.Disposed += (object obj, EventArgs args) => { };
+            IExtenderProvider[] extenderProviders = provider.GetType().GetMethod("GetExtenderProviders", BindingFlags.Instance | BindingFlags.NonPublic)
+                .Invoke(provider, new[] { testComponent }) as IExtenderProvider[];
+            IExtenderProvider result = Assert.Single(extenderProviders);
+            Assert.IsType<TestExtenderProvider>(result);
+        }
+
+        private static object CreateReflectTypeDescriptionProviderInstance()
+        {
+            Assembly srcAsssemby = Assembly.Load(Assembly.GetExecutingAssembly().GetReferencedAssemblies().FirstOrDefault(a => a.FullName.StartsWith("System.ComponentModel.TypeConverter")));
```

Comment:
We should be able to test these scenarios using public APIs. The `ReflectTypeDescriptionProvider` is the default one, so you should be able to use the normal APIs to do this work. Let's only use reflection as a last resort.